src/action/head_to_goal_control/head_to_goal_control/controller.py

```diff
@@ -90,7 +100,19 @@ def steer_to_speed(self, steering_angle):
         speed = self.max_speed * np.exp(- (steering_angle ** 2) / (2 * (self.speed_adjuster_width ** 2)))
         return speed
 
+    def desired_speed_callback(self, msg):
+        self.desired_speed = msg.data
+
+        if self.desired_speed == -1.0:
+            self.current_speed = 0.0
+            self.Pose_to_track_in_global_frame = None
+        # self.get_logger().info(f"Recieved desired speed: {self.desired_speed}")
+
     def apply_speed_decay(self):
+        if self.usingRL:
+            self.current_speed = self.desired_speed
+            return
+
         self.current_speed = (0.61 ** self.speed_decay_constant) * self.max_speed
         #self.get_logger().info(f"Speed decay applied: {self.speed_decay_constant}, set current speed to {self.current_speed}")
```

Comment:
if you are gonna comment out logging messages I recommend you changing them to .debug instead of .info so that you just need to change the logging level to debug to see all information and then cntl-f what you want